### ISSUE TITLE
esmodule has `__dirname` in `@loaders.gl/textures` file `encode-texture.ts` which causes vite to fail

### ISSUE THREAD

**Issue by jessekrubin** [visgl/loaders.gl#2004]
repro: https://github.com/jessekrubin/vite-deckgl-dirname-err-repro

**Comment by ibgreen** [comment]
At first blush, it looks like vite is not respecting package.json `browser` fields.

**Comment by jessekrubin** [comment]
I cloned and built the loaders repo. I don't think loaders.gl/textures builds correctly. 

Main and module both point to the same index.js file which looks to be use esm exports and imports as well as __dirname.

**Comment by ibgreen** [comment]
> mMain and module both point to the same index.js file which looks to be use esm exports 

Correct, the current plan is to drop ES5 builds in v4.0. If you look at `3.1-release` branch there should still be separate es5 and esm builds.

> as well as __dirname

That file is only supposed usage of __dirname looks suspect. Checking if it is needed #2005

**Comment by jessekrubin** [comment]
When I build the 3.1 branch I still get __dirname in the esm dist.

**Comment by varna** [comment]
NPM @3.1.0 `@loaders.gl/textures/dist/esm/lib/encoders/encode-texture.js`:
```js
import { ChildProcessProxy } from '@loaders.gl/worker-utils';
const MODULE_ROOT = "".concat(__dirname, "/../../..");
export async function encodeImageURLToCompressedTextureURL(inputUrl, outputUrl) {
  const args = ['texture-compressor', '--type', 's3tc', '--compression', 'DXT1', '--quality', 'normal', '--input', inputUrl, '--output', outputUrl];
  const childProcess = new ChildProcessProxy();
  await childProcess.start({
    command: 'npx',
    arguments: args,
    spawn: {
      cwd: MODULE_ROOT
    }
  });
  return outputUrl;
}
//# sourceMappingURL=encode-texture.js.map
```

**Comment by varna** [comment]
> When I build the 3.1 branch I still get __dirname in the esm dist.

Can confirm that 3.1 build does include `__dirname`

**Comment by ibgreen** [comment]
Fixed in 3.1.1: https://unpkg.com/browse/@loaders.gl/textures@3.1.1/dist/esm/lib/encoders/encode-texture.js
